### ISSUE TITLE
Evaluate Team Associated iFHSS (XP2-SSI) protocol/board

### ISSUE THREAD

**Issue by pwkpete** [pascallanger/DIY-Multiprotocol-TX-Module#943]
Pascal,

Per the other portion of discussion on RCG - a couple of us were looking to see if the Team Associated protocol could be reverse engineered based on what you see here (and or additional signal dumps if needed)

This is a single model control with bind button, no display, 2 channel with 4 pot's for EPA/trim adjustments

![PXL_20240209_212013269 MP](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/16245967/dcba5151-a7f0-4056-9495-f4681485157a)

**Comment by pascallanger** [comment]
Yes it might be possible to do since the hw is compatible with multi.
You need to connect the logic analyzer on the spi bus of the cyrf6936

**Comment by pwkpete** [comment]
You wouldn't have a link to the pinout of that chip - I can't seem to find one. From there I can backtrace to hopefully find a good solder point.

**Comment by pwkpete** [comment]
Ok, I can find

MOSI
MISO
CLK
Ground

I don't see CSN on the chip? Is that SS?

**Comment by pascallanger** [comment]
Yes SS sounds good

**Comment by pwkpete** [comment]
On the other side of the board there is a somewhat convenient debug header, but it would appear SS is not connected. I have MOSI, MISO, CLK, GND - not sure what value this would be without SS, unless SS is not what we are looking for?

From the data sheet (https://media.digikey.com/pdf/Data%20Sheets/Cypress%20PDFs/CYRF6936.pdf):
![image](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/16245967/508fb7d2-59f1-4145-a9e0-58021350e3bf)

![image](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/16245967/65e08f99-b8cf-49b2-9f24-e3c7db0c7959)

**Comment by pascallanger** [comment]
SS is really the signal we need.
I'm not in front of my PC so I can't help you following the traces

**Comment by pwkpete** [comment]
No worries, I can solder to pick up ss, but not sure why there would be a
header like this without that signal.. hmm

Will run scans tonight or tomorrow

**Comment by pascallanger** [comment]
The SPI bus can be shared with other components. This is where you need that chip select signal. May be they have something else they can connect like a screen on another version of the TX.

**Comment by pwkpete** [comment]
Gotcha.

Let me know if this looks sane, I'll grab other scans if so.

[Power On.zip](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/files/14230780/Power.On.zip)

**Comment by pwkpete** [comment]
[TX and RX connect.zip](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/files/14230872/TX.and.RX.connect.zip)

**Comment by M-i-c-h-a-e-l1** [comment]
Hello,

I had a look at two of my Team Associated radios. The first is the XP3-SS iFHSS radio from the [SC8e RTR](https://www.associatedelectrics.com/teamassociated/cars_and_trucks/SC8e/SC8e_Brushless/):

![image](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/131524236/904a6386-70d8-4c3b-b607-c82f6202ad16)

![image](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/131524236/76676779-689d-4c09-a104-09b1401d4ce9)

The electronics seem to be different than those of XP2-SSI:

![image](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/131524236/ed7a6ff4-d6bb-4780-9e0a-630d9b7b43a8)

Close-up:

![image](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/131524236/6ac84c88-fb1c-42e2-8447-26339693afc8)

The board mentions 'Thunder Tiger' as manufacturer.

The second radio is the XP120 from the [MT10 Rival RTR](https://www.associatedelectrics.com/teamassociated/cars_and_trucks/RIVAL_MT10/Brushless_RTR/):

![image](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/131524236/9d943eb1-9520-447b-a984-24453c35ab5e)

![image](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/131524236/b32eb48e-42dd-4057-899f-03c9dfdf82b4)


Close-up of the board:

![image](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/131524236/2af5a445-86dc-4bec-ab36-0a19c50bf8c7)

I am not sure if these two readios use the same protocol as the processors seem to be different.

Kind regards
Michael

**Comment by pwkpete** [comment]
I suspect the 2 and the 3 are compatible, both Cypress chips and a quick glance, the SPI signals look similar.

IIRC I saw receivers for the two listed somewhere as compatible

The 120 looks like something else, you ever try swapping them?

**Comment by RickzFPV** [comment]
I am trying to figure this out too for the SC28 line of transmitters from Team Associated.  I am realizing by following this post, and open issue that Team Associated has a ton of different Transmitters out there and all seem to be different protocols.  With all the Research I have done on it all I have been able to find is that Fcc Paperwork states it is Flysky AFHDS Protocol which it is not.  Other documentation states it is SLT.  I would really like to find out the protocol, because you can not even get a replacement transmitter from Team Associated.  Supposedly in an article I read it is the same Transmitter that the Micro X cars use.  By pictures it looks the same but if it actually is I don't Know.  Here are some pictures of it when I opened it up to take a look.

Rick
![Receiver](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/86533290/5c9d43d4-e4e1-45ee-84a1-084123d80a81)
![Transmitter 1](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/86533290/7d5337e7-797b-4671-99a4-7d8f26ab8fed)
![Transmitter 2](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/86533290/b104c273-41ce-47be-9158-c710375ab000)
![Transmitter 3](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/86533290/883d588a-75cd-46dd-b6a2-995bc29271ea)
![Transmitter 4](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/assets/86533290/537dd1f6-bf03-4eb2-a994-a8bcad400e3b)

**Comment by RickzFPV** [comment]
so is this a dead issue now with the team associated transmitters.  If someone could let me know if there is anything I can do to help with this I absolutely will

Rick

**Comment by pascallanger** [comment]
> so is this a dead issue now with the team associated transmitters. If someone could let me know if there is anything I can do to help with this I absolutely will
> 
> Rick

On the latest pictures you've posted, there is a XN297 RF chip (8 pins near the antenna). If you can solder a logic analyzer ($10 on ebay), solder it on the 3 SPI lines + GND and do a couple of dumps, I can surely add the protocol.

**Comment by M-i-c-h-a-e-l1** [comment]
What are the priorities with the implementation? The older iFHSS protocol or the newer protocol for the 1/28 scale cars?

Kind regards
Michael

**Comment by pascallanger** [comment]
I would say in whatever order the people give me things to work on. Might be dumps, send me the TX+RX or donate some money so I can purchase locally.

**Comment by pwkpete** [comment]
Was the last data file I uploaded any value? If there is something you want
me to test now, I can do that over the weekend

**Comment by M-i-c-h-a-e-l1** [comment]
Regaring your request on rcgroups Feb 20, 2024, 
[#24897](https://www.rcgroups.com/forums/showpost.php?p=52142955&postcount=24897) to make a different builds for air and surface, I see your problem with the limited flash space. 

I also use Multiplex 2,4Gz air receivers with telementry for surface applications. So for me separating air and surface would be a disadvantage as I use both air and surface on the same radio. 

Would it be possible to create a version where you just select the protocols you want to install from the list of all available protocols on Github? After coosing the protocols, the firmware with onlt the selected protocols is compiled. 

As an example: I use Multiplex 2,4Ghz air and surface, Flysky AFHDS2A air and surface. I would just need a firmware with these two protocols.

Would this be an alternative to get around the limited flash memory for the firmware? The user would then only select the protocols really needs. 

Kind regards
Michael

**Comment by pascallanger** [comment]
> Regaring your request on rcgroups Feb 20, 2024, [#24897](https://www.rcgroups.com/forums/showpost.php?p=52142955&postcount=24897) to make a different builds for air and surface, I see your problem with the limited flash space.
> I use Multiplex 2,4Gz air receivers with telementry for surface applications. So for me separating air and surface would be a disadvantage as I use both air and surface on the same radio. As an example: I use Multiplex 2,4Ghz air and surface, Flysky AFHDS2A air and surface. I would just need a firmware with these two protocols.

Multiplex like all the big protocols are included (Devo, walkera, HoTT, Hitec, flysky, frsky, multiplex...). The ground version is ready, Ben is working on the github build automation.
If I've missed any protocols we can easily add it but I think I've done a good selection.

> Would it be possible to create a tool where the user selects the protocols to install from the list of all available protocols on Github? After choosing the protocols, the firmware with only the selected protocols is compiled.

This is what you do by compiling yourself. Edit the _config.h file, select the protocols you want then compile, upload and done. This is really easy.

**Comment by M-i-c-h-a-e-l1** [comment]
Thank you for the quick reply. 

Compiling is beyond my capabilities :-( 

Is there a video tutorial on how to do this so that I can learn?

Kind regards
Michael

**Comment by pascallanger** [comment]
It's not beyond, it's actually pretty easy. Just follow each steps carefully: https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/blob/master/docs/Compiling_STM32.md

**Comment by M-i-c-h-a-e-l1** [comment]
Ok, thanks, I will have a look at it.

Kind regards
Michael

**Comment by pascallanger** [comment]
Air and Surface builds are available here: https://downloads.multi-module.org/latest-test/

**Comment by M-i-c-h-a-e-l1** [comment]
Tested the following on my desk:

Radio:
- Radiomaster Zorro
- edgeTX 2.8.0
- MM firmware 1.3.4.00

Receivers:
- Air: Multiplex RX7 with telemetry: -> **ok**
- Air: Frsky X8R -> **ok** (settings: FrSky X2 LBT EU)
- Air: Flysky FS-iA6B with telemetry: -> **ok**

- Surface: Spektrum SR3100 clone: -> **ok** (settings: DSM2, subtype 2 2F, servo 11ms)
- Surface: Reely BS-4 (=Flysky FS-BS4) with telemetry: -> **ok** (not using channel 1 as this is for the built-in gyro)
- Surface: Traxxas 6533B: protocol not supported
- Surface: Team Associated iFHSS: protocol not (yet) supported

Kind regards
Michael

**Comment by pwkpete** [comment]
I still have the tx wired up, so if there are any further tests needed, I
can run them.

**Comment by RickzFPV** [comment]
@M-i-c-h-a-e-l1  I know there are a lot of people wanting the 1/28 scale associated protocol due to broken or non working transmitters.  I  can order a logic analyzer as recommended above but i dont know if i have the knowledge to be able to use it.  I onow for the ifhss version there is a pin diagram but I can’t seem to find one for the 1/28 scale transmitter, and i don’t have the skills to figure it out on my own.

rick

**Comment by M-i-c-h-a-e-l1** [comment]
Yes, I t would be better if a different issue would be opened for this protocol as it is different.

Link to the [data sheet for the XN297](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwjX1M_I8cGEAxUIT6QEHSfGAF0QFnoECBkQAQ&url=https%3A%2F%2Fwww.panchip.com%2Fstatic%2Fupload%2Ffile%2F20190916%2F1568621331607821.pdf&usg=AOvVaw34Rj8MqAXxC5UU1WBkKxsk&opi=89978449)

Kind regards
Michael

**Comment by pwkpete** [comment]
Sorry, didn't see your reply - no worries at all. I'm ready whenever you are, just happy to have been successful with one so far!

**Comment by pascallanger** [comment]
This protocol is going to be hard work. With only one TX we won't be able to implement different IDs unless you can connect the logic analyzer to the RX too.
Is there a bind button on the TX? What's the bind procedure?
You need to do multiple dumps like we did before for each sticks, buttons and failsafe.

**Comment by pwkpete** [comment]
I can do the multiple dumps as well, just wanted to make sure the ones I uploaded looked good before going further.

I have not opened an RX so not sure how difficult that would be, however, I do happen to have a 2nd TX, if that will help...

**Comment by pascallanger** [comment]
The second TX will definitively help but I don't think it will give us the full picture.
The TX is using a large table of 48 frequencies, finding how it's generated is going to be extremely difficult if not impossible... That will limit the number of IDs. Also the SOP and DATA codes could be dependent of the ID.
There are also some packets which are different from the others which is why I would like to know the binding procedure, if there is failsafe from the radio or all other settings the radio can have.

**Comment by pascallanger** [comment]
> Sorry, forgot to answer your bind question - there is a bind button on the TX as well as the RX

So I need a capture of the bind at power on.

**Comment by M-i-c-h-a-e-l1** [comment]
@pwkpete

Could you post a photo on how you connected the radio to the analyser?

Kind regards
Michael

**Comment by pwkpete** [comment]
I just used the header in the above picture and then used one of the holes to get SS. I listed the pinouts above


Here are 2 bind captures one with IFHSS and IFHSS+

[Power+bind IFHSS-.zip](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/files/14427358/Power%2Bbind.IFHSS-.zip)
[Power +bind IFHSS+.zip](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/files/14427359/Power.%2Bbind.IFHSS%2B.zip)

Let me know then if you need captures for each IFHSS setting

**Comment by pwkpete** [comment]
@pascallanger 

Did the IFHSS setting make any difference in the scans? Do I need to run tests with each setting?

**Comment by pwkpete** [comment]
@pascallanger 

Were you able to determine if this was feasible from my last upload? If not, I can disconnect the logic probe. LMK

Thanks!
